src/components/blog/table-of-contents.tsx

```
@@ -0,0 +1,102 @@
+        </p>
+        <ul className="space-y-2 text-sm">
+          {headings.map((heading) => (
+            <li key={heading.id} className={heading.level === 3 ? "ml-4" : ""}>
+              <button
+                onClick={() => scrollToHeading(heading.id)}
+                className={`text-left transition-colors duration-200 hover:text-primary ${
+                  activeId === heading.id
+                    ? "text-primary font-medium"
+                    : "text-muted-foreground"
+                }`}
+              >
```

Biome (2.1.2)

[error] Provide an explicit type prop for the button element.

The default type of a button is submit, which causes the submission of a form when placed inside a `form` element. This is likely not the behaviour that you want inside a React application.
Allowed button types are: submit, button or reset

(lint/a11y/useButtonType)